fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaClient.java

```diff
@@ -668,7 +668,7 @@ public TextFileFormatDesc getTextFileFormatDesc(StorageDescriptor sd) {
 
     public List<HdfsFileDesc> getHdfsFileDescs(String dirPath, boolean isSplittable,
                                                StorageDescriptor sd) throws Exception {
-        URI uri = new URI(dirPath);
+        URI uri = new Path(dirPath).toUri();
```

Comment:
I think you'd better find the core part of URI source code

Reply:
@stephen-shelby 【new Path(dirPath)】will replace space character  with "%20" 
detail function at: java.net.URI#appendEscape